modules/file/file.go

```diff
@@ -1478,6 +1516,72 @@ func (r *run) checkHash(f fileEntry, hashtype checkType) {
 	return
 }
 
+func (r *run) checkBytes(f fileEntry) {
+	var (
+		err error
+	)
+	defer func() {
+		if e := recover(); e != nil {
+			err = fmt.Errorf("checkBytes() -> %v", e)
+		}
+	}()
+	reader := f.getReader()
+	defer f.Close()
+
+	// Must use the "label,search" for loop iteration because of
+	// map[string]search for parameters
+	for _, search := range r.Parameters.Searches {
+		if !search.isactive {
+			continue
+		}
+
+		for _, c := range search.checks {
+
+			decodedByte := make([]byte, hex.DecodedLen(len(c.bytes)))
+			_, err := hex.Decode(decodedByte, c.bytes)
+			if err != nil {
+				e := fmt.Sprintf("unable to decode byte query: %v ", err)
+				panic(e)
+			}
+			scansize := len(decodedByte) // Get the actual byte slice length
+			blocksize := 4 * (2 << 10)   // 4096, 4k
+
+			var bigbuf = make([]byte, (blocksize + (2 * scansize)))
+			// Read slightly more than 4k so we have a full buffer to scan through
+			// as we move into the for loop
+			_, err = reader.Read(bigbuf)
+			if err != nil && err != io.EOF {
+				panic(err)
+			}
+
+			for err != io.EOF {
+				if bytes.Contains(bigbuf, decodedByte) {
+					// We do find the intended hex values in our scan
+					// Break the loop and return
+					// Not certain exactly how to do that here
```

Comment:
I think what we would want to do is mark the check status as true if it has matched (e.g., checksstatus[label][i] = true and continue reading the file, so the other checks that may need to run can execute?